src/foam/core/Property.js

```diff
@@ -205,6 +205,29 @@ foam.CLASS({
           return comparePropertyValues(f(o1), f(o2));
         };
       }
+    },
+    {
+      name: 'diffPropertyValues',
+      transient: true,
+      value: function(v1, v2, diff) {
+        if ( Array.isArray(v1) ) {
+          var subdiff = foam.Array.diff(v1, v2);
```

Comment:
Would be better to not perform this check here but instead have a different implementation of diffPropertyValues in Array Property classes, but I'll accept this now because it's still better than what we have.